diff -ru xmlrpc++0.7.orig/src/XmlRpc.h xmlrpc++0.7/src/XmlRpc.h
--- xmlrpc++0.7.orig/src/XmlRpc.h	2003-03-06 10:12:28.000000000 -0600
+++ xmlrpc++0.7/src/XmlRpc.h	2020-05-08 19:35:13.650119626 -0500
@@ -25,6 +25,7 @@
 # include <string>
 #endif
 
+#include <string.h>
 #include "XmlRpcClient.h"
 #include "XmlRpcException.h"
 #include "XmlRpcServer.h"
diff -ru xmlrpc++0.7.orig/src/XmlRpcUtil.h xmlrpc++0.7/src/XmlRpcUtil.h
--- xmlrpc++0.7.orig/src/XmlRpcUtil.h	2003-03-06 09:27:14.000000000 -0600
+++ xmlrpc++0.7/src/XmlRpcUtil.h	2020-05-08 19:35:13.606119992 -0500
@@ -20,6 +20,7 @@
 # define strcasecmp stricmp
 # define strncasecmp strnicmp
 #endif
+#include <string.h>
 
 namespace XmlRpc {
 
diff -ru xmlrpc++0.7.orig/test/Validator.cpp xmlrpc++0.7/test/Validator.cpp
--- xmlrpc++0.7.orig/test/Validator.cpp	2003-01-08 15:55:40.000000000 -0600
+++ xmlrpc++0.7/test/Validator.cpp	2020-05-08 19:35:14.278114392 -0500
@@ -137,7 +137,7 @@
   {
     std::cerr << "ModerateSizeArrayCheck\n";
     std::string s = params[0][0];
-    s += params[0][params[0].size()-1];
+    s += std::string(params[0][params[0].size()-1]);
     result = s;
   }
 } moderateSizeArrayCheck(&s);
Only in xmlrpc++0.7/test: Validator.cpp~
