diff -ru xmlrpc++0.7.orig/makefile xmlrpc++0.7/makefile
--- xmlrpc++0.7.orig/makefile	2003-03-06 11:28:48.000000000 -0600
+++ xmlrpc++0.7/makefile	2020-05-08 20:13:31.490983426 -0500
@@ -1,11 +1,11 @@
 # makefile written for gnu make
 CXX		= g++
 SRC		= ./src
-CPPFLAGS	= -I$(SRC)
+CPPFLAGS	= -I$(SRC) -m32
 DEBUG		= -g
 OPTIMIZE	= -O2
 GCCWARN		= -Wall -Wstrict-prototypes
-CXXFLAGS	= $(DEBUG) $(GCCWARN) $(OPTIMIZE) $(INCLUDES)
+CXXFLAGS	= $(DEBUG) $(GCCWARN) $(OPTIMIZE) $(INCLUDES) -m32
 
 LIB		= ./libXmlRpc.a
 
diff -ru xmlrpc++0.7.orig/src/XmlRpc.h xmlrpc++0.7/src/XmlRpc.h
--- xmlrpc++0.7.orig/src/XmlRpc.h	2003-03-06 10:12:28.000000000 -0600
+++ xmlrpc++0.7/src/XmlRpc.h	2020-05-08 20:11:59.311749605 -0500
@@ -25,6 +25,7 @@
 # include <string>
 #endif
 
+#include <string.h>
 #include "XmlRpcClient.h"
 #include "XmlRpcException.h"
 #include "XmlRpcServer.h"
diff -ru xmlrpc++0.7.orig/src/XmlRpcUtil.h xmlrpc++0.7/src/XmlRpcUtil.h
--- xmlrpc++0.7.orig/src/XmlRpcUtil.h	2003-03-06 09:27:14.000000000 -0600
+++ xmlrpc++0.7/src/XmlRpcUtil.h	2020-05-08 20:11:59.311749605 -0500
@@ -20,6 +20,7 @@
 # define strcasecmp stricmp
 # define strncasecmp strnicmp
 #endif
+#include <string.h>
 
 namespace XmlRpc {
 
diff -ru xmlrpc++0.7.orig/test/Validator.cpp xmlrpc++0.7/test/Validator.cpp
--- xmlrpc++0.7.orig/test/Validator.cpp	2003-01-08 15:55:40.000000000 -0600
+++ xmlrpc++0.7/test/Validator.cpp	2020-05-08 20:11:59.311749605 -0500
@@ -137,7 +137,7 @@
   {
     std::cerr << "ModerateSizeArrayCheck\n";
     std::string s = params[0][0];
-    s += params[0][params[0].size()-1];
+    s += std::string(params[0][params[0].size()-1]);
     result = s;
   }
 } moderateSizeArrayCheck(&s);
